### PR TITLE
Use pathlib to determine location of module

### PR DESCRIPTION
**Pull request by jdahm**
## Description

Allows for relocatable caches.

Thanks to @mcgibbon for the contribution!

Blocked by: #647.

### CODE REVIEW

File: src/gt4py/backend/module_generator.py [hunk 1]
```diff
@@ -465,11 +465,15 @@ def _is_not_empty(self) -> bool:
     def generate_imports(self) -> str:
         source = ["from gt4py import utils as gt_utils"]
         if self._is_not_empty():
+            assert self.pyext_file_path is not None
+            file_path = "f{{pathlib.Path(__file__).parent.resolve()}}/{}".format(
+                os.path.basename(self.pyext_file_path)
+            )
```
Comment (egparedes) [inline src/gt4py/backend/module_generator.py:471]: It looks like there is an error here. Should this be something like:
```suggestion
            file_path = "{{pathlib.Path(__file__).parent.resolve()}}/{base}".format(
                base=os.path.basename(self.pyext_file_path)
            )
```
?

**Review by egparedes — COMMENTED** [review]
The idea looks good, but the implementation contains some kind of bug.

File: src/gt4py/backend/module_generator.py (hunk 1 above)
Comment (jdahm) [reply src/gt4py/backend/module_generator.py]: The `f` has to be pulled outside the quotes. I think I got it now. The latest commit I pushed works locally.

**Comment by egparedes** [comment]
borg try

**Comment by egparedes** [comment]
bors try

**Comment by jdahm** [comment]
bors try

File: src/gt4py/backend/module_generator.py [hunk 2]
```diff
@@ -465,11 +465,15 @@ def _is_not_empty(self) -> bool:
     def generate_imports(self) -> str:
         source = ["from gt4py import utils as gt_utils"]
         if self._is_not_empty():
+            assert self.pyext_file_path is not None
+            file_path = 'f"{{pathlib.Path(__file__).parent.resolve()}}/{}"'.format(
+                os.path.basename(self.pyext_file_path)
+            )
```
Comment (egparedes) [inline src/gt4py/backend/module_generator.py:471]: Mixing escaped and non-escaped placeholders is a bit confusing. I propose to replace the `format()`  call by a simple string concatenation:
```suggestion
            file_path = 'f"{{pathlib.Path(__file__).parent.resolve()}}/"' + str(
                os.path.basename(self.pyext_file_path)
            )
```

**Comment by jdahm** [comment]
This build passed, but it failed to propagate the success back to github. It just needs your re-review to merge.

**Pull request merged**